lib/cea/sei_processor.js

```diff
@@ -0,0 +1,122 @@
+/*! @license
+ * Shaka Player
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+goog.provide('shaka.cea.SeiProcessor');
+
+
+/**
+ * H.264 SEI NALU Parser used for extracting 708 closed caption packets.
+ */
+shaka.cea.SeiProcessor = class {
+  constructor() {
+    /**
+     * Length of the buffered SEI data.
+     * @private {number}
+     */
+    this.seiLength_ = 0;
+
+    /**
+     * Buffer for the supplemental enhancement information (SEI) network
+     * abstraction layer unit (NALU).
+     * @private @const {!Uint8Array}
+     */
+    this.seiArray_ = new Uint8Array(5 * 1024);
```

Comment:
Why do you preallocate a 5k array for this?

Each call to process() is proceeded by a call to append(), and process()  calls clear() at the end.  So why not just operate on the data from the caller directly instead of copying it to another buffer and then wiping it out again?

Reply:
This is more relevant to a potential future TS Parser in which copying to another buffer may be necessary.

I've removed this overhead for now and done what you said, which makes perfect sense. One thing to note with caution is that the call to subarray() in process() returns a reference to the original array rather than a copy, so we need to be careful operating on this array so as not to modify the original array. But this is more a note for me when implementing my decoder :)